### PR TITLE
Hotflix/team members

### PR DESCRIPTION
**Pull request by eman-ijaz7vals**

### CODE REVIEW

**Comment by eman-ijaz7vals** [comment]
Cycle-0 Review done

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -19,6 +20,7 @@ def new
   # GET /show
   def show
     @team_memberships, @all_members = @team.member_and_team_memberships
+    @pendings = @team.pending_requests(@team.id)
```
Comment (abubakarazeem) [inline app/controllers/teams_controller.rb]: no need to pass `id` in this case

File: app/controllers/teams_controller.rb [hunk 2]
```diff
@@ -85,13 +87,13 @@ def destroy
 
   def add_membership
     @is_admin = params[:join_admin][:joining_decision] != '0'
-    @team_membership = TeamMembership.new(is_team_admin: @is_admin, is_approved: params[:is_approved], team_id: params[:team_id], user_id: params[:user_id])
-    respond_to do |format|
-      if @team_membership.save
+    @team_membership = @team.team_memberships.new(is_team_admin: @is_admin, is_approved: params[:is_approved], team_id: params[:team_id], user_id: params[:user_id])
```
Comment (abubakarazeem) [inline app/controllers/teams_controller.rb]: use build instead of new

File: app/controllers/teams_controller.rb [hunk 3]
```diff
@@ -85,13 +87,13 @@ def destroy
...
+    if @team_membership.save
+      respond_to do |format|
         format.js
-      else
-        flash.now[:error] = @team_membership.errors.full_messages
       end
+    else
+      flash.now[:error] = @team_membership.errors.full_messages
```
Comment (abubakarazeem) [inline app/controllers/teams_controller.rb]: respond_to is missing in 2nd case when team not created..

File: app/helpers/teams_helper.rb [hunk 4]
```diff
@@ -2,4 +2,7 @@ module TeamsHelper
   def requested(team_id, user_id)
```
Comment (abubakarazeem) [inline app/helpers/teams_helper.rb]: `?` is missing in method name

File: app/helpers/teams_helper.rb [hunk 5]
```diff
@@ -2,4 +2,7 @@ module TeamsHelper
...
     TeamMembership.find_by(team_id: team_id, user_id: user_id, is_approved: false).present?
   end
+  def already_joined_team(team_id, user_id)
```
Comment (abubakarazeem) [inline app/helpers/teams_helper.rb]: `?` missing in method name

File: app/models/team.rb [hunk 6]
```diff
@@ -20,4 +20,7 @@ def member_and_team_memberships
     @all_members = User.where.not(id: already_present_member_ids).select(:id, :first_name)
     [@team_memberships, @all_members]
   end
+  def pending_requests(id)
```
Comment (abubakarazeem) [inline app/models/team.rb]: no need to pass id.. we can use `id` method

File: app/views/teams/_add_member_modal.html.erb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        <h5 class="modal-title" id="exampleModalLabel">Add Member</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <%= form_tag add_membership_team_path(@team.sequence_num), method: 'get', remote: true do %>
+        <div class="modal-body" id="add_member_modal" >
+          <div class="row">
+            <div class="col-8">
+              <%= select_tag "user_id", options_from_collection_for_select(@all_members, :id, :first_name), id: 'js_members_select', class: 'select_tag_width'%>
```
Comment (abubakarazeem) [inline app/views/teams/_add_member_modal.html.erb]: don't use `js` in  id of an element

File: app/views/teams/add_membership.js.erb [hunk 8]
```diff
@@ -1,2 +1,3 @@
-alert('Member added');
+alert("membership added")
```
Comment (abubakarazeem) [inline app/views/teams/add_membership.js.erb]: optional: use flash messages

File: app/views/teams/index.html.erb [hunk 9]
```diff
@@ -1,25 +1,27 @@
...
-      <%= link_to 'New Team', new_team_path,class: 'btn btn-secondary btn-sm' %>
-    <% end %>
-  </div>
+<h1 class="float-left">Teams</h1>
+<% if can? :create, Team %> 
+  <button type="button" class="btn btn-secondary float-right mr-2" data-toggle="modal"data-target="#create_team_modal">Create Team
+  </button>
+  <%= render partial: 'create_team_modal' %>
+<% end %>
+<br><br><br>
```
Comment (abubakarazeem) [inline app/views/teams/index.html.erb]: optional: use margin instead of `<br>`

File: app/views/teams/show.html.erb [hunk 10]
```diff
@@ -1,24 +1,27 @@
...
-
   <div>
     <h3>Members </h3>    
     <table class="table table-hover" id="team_members_table">
       <thead>
         <th scope="row">Name</th>
         <th scope="row">Admin</th>
-        <th scope="row">Action</th>
+        <% if can? :manage, Team %>
+         <th scope="row">Action</th>
```
Comment (abubakarazeem) [inline app/views/teams/show.html.erb]: indentation

File: app/assets/stylesheets/custom.css.scss [hunk 11]
```diff
@@ -5,3 +5,15 @@
 .upload_document_field {
   width: 70%;
 }
+
```
[the remark this answers is not in the document]
Comment (eman-ijaz7vals) [reply app/assets/stylesheets/custom.css.scss]: upload_document_field styling is related to projects and issues, not to teams

**Pull request merged**